pkg/webhooks/ingresscontroller/ingresscontroller.go

```diff
@@ -149,12 +176,151 @@ func (wh *IngressControllerWebhook) authorized(request admissionctl.Request) adm
 		}
 	}
 
-	ret = admissionctl.Allowed("IngressController operation is allowed")
+	/* TODO:
+	 * 1) ONLY check for privatelink clusters? How?
+	 * 2) HCP vs Classic, etc.. ...any other cluster type this should apply to? Is this handled by the
+	 * HypershiftEnabled vs ClassicEnabled flags set in this module? Currently HCP disabled.
+	 * If HCP is to be enabled for allowed source ranges, should this part of a 2nd ingress validator to
+	 * allow separation of validations between cluster install types? Is there a run time method available
+	 * to this validator to determine classic vs hcp?
```

Comment:
> HCP vs Classic, etc.. ...any other cluster type this should apply to?
Probably OpenShift Dedicated as well (unless that's covered under `Classic`)

>Is this handled by the * HypershiftEnabled vs ClassicEnabled flags set in this module? Currently HCP disabled.
I'm not sure of the impact of this 🤔

> If HCP is to be enabled for allowed source ranges, should this part of a 2nd ingress validator to allow separation of validations between cluster install types?
This validator is _only_ needed on the `default` Ingress Controller

Reply:
Thanks.  Updated this comment with info/dialog from OSD-24275 as well. 
The ingresscontroller webhook currently has HypershiftEnabled set to false, and ClassicEnabled set to true. I think additional dialog in OSD-24275 suggests we'll likely want to separate HCP efforts and webhook to be tracked under a different Jira.